packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/computeAxisValue.ts

```diff
@@ -208,6 +210,19 @@ export function computeAxisValue<T extends ChartSeriesType>({
       scale: finalScale.domain(domain) as any,
       tickNumber,
       colorScale: axis.colorMap && getColorScale(axis.colorMap),
+      valueFormatter:
+        axis.valueFormatter ??
+        (createScalarFormatter(
+          tickNumber,
+          getScale(
+            scaleType,
+            range.map((v) => scale.invert(v)),
+            range,
+          ),
+        ) as <TScaleName extends ScaleName>(
+          value: any,
+          context: AxisValueFormatterContext<TScaleName>,
+        ) => string),
```

Comment:
We could make this a bit more readable by moving it our of the obj assignment and using the direct type
```ts
const scalarFormatter = createScalarFormatter(
  tickNumber,
  getScale(
    scaleType,
    range.map((v) => scale.invert(v)),
    range,
  ),
) as AxisConfig['valueFormatter'];

...
valueFormatter: axis.valueFormatter ?? scalarFormatter,
```

Reply:
> Yeah, it's more readable. I didn't do it to avoid creating the formatter if it isn't needed.

It is not like we are processing anything heavy in this step 😆

Reply:
My reasoning was that `computeAxisValue` can potentially be called on every frame when zooming, so we probably want to keep it as light as possible. This is unlikely to have a meaningful impact, but if we always use that reasoning we'll die by a thousand performance cuts 😛